library/std/src/thread/local.rs

```diff
@@ -1130,7 +1130,7 @@ pub mod os {
         unsafe {
             let ptr = Box::from_raw(ptr as *mut Value<T>);
             let key = ptr.key;
-            key.os.set(1 as *mut u8);
+            key.os.set(ptr::invalid_mut(1));
```

Comment:
Shouldn't this be dangling instead? Is there semantically a difference between those two?

Reply:
While We All Know how NonNull is implemented, technically it does not guarantee the value of its address:

> Creates a new NonNull that is dangling, but well-aligned.

I opted to preserve the precise behaviour/semantics here and use the exact requested value.

Reply:
Ah, that makes sense. In this case, although I don't think the whole module needs to be refactored, it probably would have made sense to assign the 1 to a constant.

This is why all those "don't use magic numbers except for 0, 1, or 2" rules don't always apply.